### ISSUE TITLE
Email addresses show up TWICE in listview but NOT in single view

### ISSUE THREAD

**Issue by jobst** [salesagility/SuiteCRM#8797]
## Issue
In Listviews email addresses for contacts show up twice, ONCE as a link than without space a second time without being a link like so:

`<a class="email-link" href="mailto:JohnDoe@example.com">JohnDoe@example.com</a>JohnDoe@example.com`

I traced the creation of the email address in the function 

`private function createEmailLink($module_name, $record_id, $name, $addr, $text) `

and I fiddled with the email address but it always comes up twice - this indicated that a function AFTER the execution of this codes creates the extra email address TEXT (not link).

Now I am not sure whether it is a problem related to my sugar install, I do not know.
I have repaired everything, I have turned every extra module off - still the email addresses come up twice.

#### Expected Behavior

A single show of the email address with no extra text:

`<a class="email-link" href="mailto:JohnDoe@example.com">JohnDoe@example.com</a>`

#### Actual Behavior
A double show of the email address once as linke once as text:

`<a class="email-link" href="mailto:JohnDoe@example.com">JohnDoe@example.com</a>JohnDoe@example.com`

I check the logs, there is no indication of strange behaviour.

#### Possible Fix
No idea

#### Steps to Reproduce
1.Just select any listview
2.check the email addresses

#### Context
Annoying for the people who use it, I am the admin, so I do not deal with that side of things but I feel for the people.

#### Your Environment

* SuiteCRM Version used: 7.11.13
* Browser name and version: happens in both latest chrome AND latest firefox
* Environment name and version (e.g. MySQL, PHP 7):7.3.12
* Operating System and version (e.g Ubuntu 16.04):CentOS 7.X

**Comment by samus-aran** [comment]
Can I confirm what mail client you are using on your user's profile? Is it the Internal mail client or External? It should be a dropdown option.

**Comment by jobst** [comment]
Hi Ashley

> Can I confirm what mail client you are using on your user's profile? Is it the Internal mail client or External? It should be a dropdown option.

This has nothing to do with the email client ... please let me explain. Took me a very looooong time to work this out, especially it ONLY showed up after the upgrade from 7.10 to 7.11 (never before!!) - this made it even more difficult as I assumed (rightly or wrongly) this was related to THAT upgrade.

The install I have is based on SugarCRM which I started using in 2004 (yep, that long).
I have lots of custom fields for Accounts, Contacts, Leads and other. I have a few private modules, I have also lots of external stuff which accesses SuiteCRM through a variety of Interfaces ... so it's not easy to do a fresh install, then hook the old database back into it.

The 2nd email showed up DIRECTLY behind the correct email due to do some code left by EITHER ME or SugarCRM, I have no clue whether it was a custom code add on by me or by SugarCRM.

In custom/modules/Leads/metadata/listviewdefs.php I found the following code (the dots indicate all the other stuff)
```
$listViewDefs ['Leads'] =
array (
  ....
  'EMAIL2' =>
  array (
    'type' => 'varchar',
    'studio' => 'false',
    'label' => 'LBL_LIST_EMAIL_ADDRESS',
    'width' => '15%',
    'customCode' => '{$EMAIL2_LINK}{$EMAIL2}</a>',
    'default' => false,
  ),
  ....
  'EMAIL1' =>
  array (
    'width' => '16%',
    'label' => 'LBL_LIST_EMAIL_ADDRESS',
    'sortable' => false,
    'customCode' => '{$EMAIL1_LINK}{$EMAIL1}</a>',
    'default' => true,
  ),
  ....
```
Same for Contacts, Accounts etc.

I also found language/metadata/other files that dated back pre 2015, old modules that were still hanging around, e.g. "AM_ProjectHolidays" or  "jjwg_Maps" - I spend a day cleaning/deleting/wiping those.

So once I deleted this it was all good.

I can't be the only one who has a SuiteCRM install that dates back to a SugarCRM install pre 2015.
Not sure whether it would be an idea to create a "cleaner module" or sth from your end?

Sorry I forgot to follow up on this entry but the Covid19 stuff has put us into emergency mode working 15 hours a day to keep our company alive ...

Please do not hesistate to contact me if you need more info about this.

**Comment by pgorod** [comment]
I also have this problem in an old installation.

The bug wasn't showing lately, but then after a "Rebuild Dashlets" it appeared.

I found I had this file: **`custom/modules/Leads/metadata/dashletviewdefs.php`**

containing this part:

```
  'email1' =>
  array (
    'width' => '30%',
    'label' => 'LBL_EMAIL_ADDRESS',
    'sortable' => false,
    'customCode' => '{$EMAIL1_LINK}{$EMAIL1}</a>',
    'default' => true,
    'name' => 'email1',
  ),
```

And I just needed to remove that bit that says **`{$EMAIL1}`** and now it looks ok, no double address is shown.

I don't think this is a bug, it's just a rather unfortunate combination of fixes to core, versus old custom files. So I am closing the issue, but if you disagree, let me know, we can re-open. Thanks
